pve2_api.class.php

```diff
@@ -251,20 +261,21 @@ private function action ($action_path, $http_method, $put_post_parameters = null
 					return $action_response_array['data'];
 				}
 			} else {
-				error_log("This API Request Failed.\n" .
-					"HTTP Response - {$split_http_response_line[1]}\n" .
-					"HTTP Error - {$split_headers[0]}");
+				throw new PVE2_Exception("PVE2 API: This API Request Failed.\n" .
+					"HTTP CODE: {$split_http_response_line[1]},\n" .
+					"HTTP ERROR: {$split_headers[0]},\n" . 
+					"REPLY INFO: {$body_response}");
 				return false;
```

Comment:
Can't throw _and_ return at the same time. I'd switch this back to logging and put it behind the config option I mentioned above, so the rest of the library continues working as expected.

Reply:
I don't know what I was thinking, but I would _not_, at this point, switch back to `error_log` (though proper PSR-3 would be even better than throwing exceptions).